sql/src/test/java/io/crate/action/sql/SessionTest.java

```diff
@@ -98,6 +104,42 @@ public void testParameterTypeExtractor() {
         typeExtractor.getParameterTypes(analyzedRelation);
     }
 
+    @Test
+    public void testGetParamType() {
+        Analyzer analyzer = Mockito.mock(Analyzer.class);
+        Mockito.when(analyzer.unboundAnalyze(any(), any(), any()))
+            .thenReturn(Mockito.mock(QueriedRelation.class));
+        Planner planner = Mockito.mock(Planner.class);
+        Executor executor = Mockito.mock(Executor.class);
+        Session.ParameterTypeExtractor parameterTypeExtractor = Mockito.mock(Session.ParameterTypeExtractor.class);
+        Mockito.when(parameterTypeExtractor.getParameterTypes(any())).thenReturn(
+            new DataType[] { DataTypes.INTEGER, DataTypes.INTEGER }
+        );
+
+        Session session = new Session(
+            analyzer,
```

Comment:
Btw creating a `SqlExecutor` would be an easy way to get a real analyzer/planner - using that should have the advantage that it's less likely to have to update the test if internals change.

Reply:
It would also have the advantage that the parameterTypeExtractor doesn't have to be mocked if the statement is changed to something like `select  1 + ? + ?` - so the test covers more production code

Reply:
Thanks, using `SqlExecutor` now as you proposed.